### PR TITLE
Linked Time: Update prospective fob styling

### PR DESCRIPTION
**Pull request by rileyajones**
## Motivation for features / changes
1) The text in the prospective fob was off center
2) The prospective fob "did not feel clickable"
3) Clicking near the prospective fob would enable it which lead to some weird bugs

## Technical description of changes
I changed the padding on the fob when the `allowRemoval` flag is set to fix the text centering issuel

The prospective fob is shown based on the value of the `prospectiveStep` variable which is set when hovering the `prospective-fob-area` and set to `null` when the cursor leaves that area. I moved the prospectiveFob inside the `prospective-fob-area` so the cursor would not count as leaving it when the fob appears and takes focus. I also moved the click handler to the fob object (rather than the `prospective-fob-area`) to ensure click events are more accurate.

The changes to the HTML nesting made css changes much easier allowing the fob change opacity when hovered to draw additional attention to it and the cursor should not properly change.

## Screenshots of UI changes
Before:
image

After:
Hovering Area (Not fob)
image

Hovering Fob
image

## Detailed steps to verify changes work correctly (as executed by you)
1) Start tensorboard
2) Go to http://localhost:6006?enableDataTable=true&allowRangeSelection=true&enableProspectiveFob=true#timeseries
3) Hover the X Axis of the scalar card and assert a prospective fob appears
4) Hover the prospective fob and assert the opacity and cursor change
5) Click the prospective fob and assert a start fob is added
6) Repeat steps 3-5 and assert an end fob is added
7) Hover the XAxis and assert no prospective fob is shown

## Alternate designs / implementations considered
I considered increasing the size of the `prospective-fob-area` and I also considered making the prospective fob larger when hovered.

### CODE REVIEW

File: tensorboard/webapp/widgets/card_fob/card_fob_component.scss [hunk 1]
```diff
@@ -19,20 +19,43 @@ limitations under the License.
   display: inline-block;
 }
 
+$FOB_BACKGROUND: mat.get-color-from-palette(mat.$gray-palette, 300);
+
 .fob {
   display: inline-flex;
-  background-color: mat.get-color-from-palette(mat.$gray-palette, 300);
+  background-color: $FOB_BACKGROUND;
```
Comment (japie1235813) [inline tensorboard/webapp/widgets/card_fob/card_fob_component.scss:26]: why make it an variable while it's only used once?

**Review by japie1235813 — COMMENTED** [review]
Generally lgtm, just the vertical line seems gone because of html restructure.
Screenshot 2022-11-21 at 2 30 07 PM
Since fob opacity change on hover is no longer the case, can we simply revert this part?

Another small bug is the pin still has button css behavior (hover to change the background color)
Screenshot 2022-11-21 at 2 32 11 PM
This could be fixed in separate PR.

I also want to make the text color lighter (it's too black) but I think I can adjust myself in another PR.

File: tensorboard/webapp/widgets/card_fob/card_fob_component.scss (hunk 1 above)
Comment (rileyajones) [reply tensorboard/webapp/widgets/card_fob/card_fob_component.scss]: I'll revert this.

**Comment by rileyajones** [comment]
I fixed the issue with the extended line (it was surprisingly tricky). I think I generally prefer the html restructure as it seems more intuitive to me and it is easier to refactor.

**Comment by japie1235813** [comment]
Separating the line and the fob is actually counter intuitive to me. I also don't feel setting the height 30px more just to be 30px less later is a good strategy. 
I'm not sure how much refactor needs this structure change but, isolate the extended-line element makes me feel this part need refactor later.

**Comment by japie1235813** [comment]
I double that structure is necessary for the fob to be clickable. For time's sake I'll approve this PR but I really want to look into this later.

**Review by japie1235813 — APPROVED** [review]

**Pull request merged**